roqoqo/src/operations/bosonic_operations.rs

```diff
@@ -0,0 +1,206 @@
+// Copyright © 2021-2022 HQS Quantum Simulations GmbH. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the
+// License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashSet;
+
+use crate::operations::{
+    ImplementedIn1point6, InvolveModes, InvolveQubits, InvolvedClassical, InvolvedModes,
+    InvolvedQubits, Operate, OperateModeGate, OperateSingleMode, OperateSingleModeGate,
+    OperateTwoMode, OperateTwoModeGate, Substitute, SubstituteModes, SupportedVersion,
+};
+use crate::RoqoqoError;
+use qoqo_calculator::CalculatorFloat;
+
+/// The single-mode squeezing gate with tunable squeezing.
+///
```

Comment:
We don't want to put in full latex math but a more detailed description would be helpful, also we could link an archive paper with a fitting descritption